RUFAS/biophysical/animal/ration/ration_manager.py

```diff
@@ -31,6 +33,7 @@ class RationManager:
     ration_feeds: dict[AnimalCombination, list[RUFAS_ID]] | None
     user_defined_rations: dict[AnimalCombination, dict[RUFAS_ID, float]] | None
     tolerance: float | None = 0.0
+    maximum_ration_reformulation_attempts: int = 250
```

Comment:
I think this would be better served as an input as it's something a user may want to change down the line. It seems like something that theoretically people might want to test ration formulation limits with and making it an input would allow SA work to be done with it.